app/views.py

```diff
@@ -201,3 +274,35 @@ def default(self, thing):
 
 def jsonize(x):
     return json.dumps(x, cls=DBModelJSONEncoder)
+
+# ----------------------------------------------------------- utility fns --- #
+
+def mkdirp(p):
+    if os.path.exists(p) and os.path.isdir(p):
+        return False
+    else:
+        os.makedirs(p)
+        return True
+
+
+def upload_dir(p):
+    if isinstance(p, basestring):
+        add_path = (p,) # Turn the single element into a tuple that can be
+                        # joined into a full path
+    elif isinstance(p, (list, tuple)):
```

Comment:
Perhaps,

'''
elif hasattr(p, '__iter__'):
    p = tuple(p)
'''

is more generic?